docarray/array/array/pushpull/jinaai.py

```diff
@@ -0,0 +1,345 @@
+import json
+import logging
+import os
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Type, Union
+
+import hubble
+from hubble import Client as HubbleClient
+from hubble.client.endpoints import EndpointsV2
+
+from docarray.array.array.pushpull import PushPullLike, __cache_path__
+from docarray.array.array.pushpull.helpers import (
+    _BufferedCachingRequestReader,
+    get_version_info,
+    raise_req_error,
+)
+
+if TYPE_CHECKING:  # pragma: no cover
+    import io
+
+    from docarray import BaseDocument, DocumentArray
+
+
+def _get_length_from_summary(summary: List[Dict]) -> Optional[int]:
+    """Get the length from summary."""
+    for item in summary:
+        if 'Length' == item['name']:
+            return item['value']
+    raise ValueError('Length not found in summary')
+
+
+def _get_raw_summary(self: 'DocumentArray') -> List[Dict[str, Any]]:
+    items: List[Dict[str, Any]] = [
+        dict(
+            name='Type',
+            value=self.__class__.__name__,
+            description='The type of the DocumentArray',
+        ),
+        dict(
+            name='Length',
+            value=len(self),
+            description='The length of the DocumentArray',
+        ),
+        dict(
+            name='Homogenous Documents',
+            value=True,
+            description='Whether all documents are of the same structure, attributes',
+        ),
+        dict(
+            name='Fields',
+            value=tuple(self[0].__class__.__fields__.keys()),
+            description='The fields of the Document',
+        ),
+        dict(
+            name='Multimodal dataclass',
+            value=True,
+            description='Whether all documents are multimodal',
+        ),
+    ]
+
+    return items
+
+
+class PushPullJAC(PushPullLike):
```

Comment:
why do u need to inherit from the protocol here?

Reply:
Yea you are right. This isn't needed. I didn't know this until later and accidentally left this here.